### ISSUE TITLE
Colon bug?

### ISSUE THREAD

**Issue by esraiak** [JuliaEditorSupport/julia-vim#122]
The following code,
-----
for i in 1:2
  a = b[i,:]
end
-----,
makes the syntax mark "end" in red. Is this a bug?

**Comment by carlobaldassi** [comment]
I can't reproduce this, for me it works fine.
Are you sure you have the latest version of the plugin?
Is there any other code above this which may explain the error? E.g., an open bracket would make the following `for` be part of a comprehension, which does not require an `end`, like this:
```
[something
for i in 1:2
    a = b[i,:]
end              # now this is an error
```

**Comment by esraiak** [comment]
Oops, I did not think about the fact that I'm doing this over SSH, which is necessary for me to reproduce the bug. I'm not sure whether this is particular to julia-vim (other than colons, the highlighting works fine).

So I have my laptop A and a server B. 
* julia-vim works fine on A
* also fine when SSH'ing from A to itself
* fails as above when SSH'ing from A to B.
* I cannot log in to B any other way than by SSH (openssh)
* the $TERM on both A and B are "xterm-256color". Both run bash .

Seems like it is an issue with the setup on B. The julia-vim installation on A and B are the same, cloned from github a week ago.

**Comment by carlobaldassi** [comment]
Yes, this must be an issue with the setup on B, although I can't imagine what the problem could be. I don't think ssh should have much to do with it. It's rather hard to debug this however. My suggestion is: can you obtain a list of differences between the configuration in the two machines? Mostly vimrc and plugins installed. Ideally, if you could try to clone A's configuration on B (even just temporarily) and see whether it works, this could help a lot.
